netket/hilbert/continuous_space_particles.py

```diff
@@ -0,0 +1,69 @@
+from typing import Tuple
+
+from .abstract_hilbert import AbstractHilbert
+
+class Particles(AbstractHilbert):
+    """Class for the Hilbert space of particles
+    in continuous space.
+    This class defines the common interface that
+    can be used to interact with particles defined
+    in continuous space.
+    """
+
+    def __init__(self, N: int, L: Tuple[float, ...], pbc: Tuple[bool, ...], ptype: str):
```

Comment:
Possible signatures that we should consider : 
```python
Particle(N: int =1, L: Optional[Tuple[float,...]] = None, pbc: Optional[Tuple[bool, ...]=None, type: Optional[str]=None)
```

```python
Particle(N: int = 1, Geometry: Optional[GeoType]=None, type: Optional[str]=None) 
```

personally I prefer the second, since I don't like putting too many details on the simulation box in the particle itself, since that's rather a geometry concept?

Reply:
I like the second better, but maybe we could have `L=Tuple[float,...]` be a shortcut to construct the corresponding `L1xL2xL3..` geometry?
I'd refrain from defining the `GeoType` interface now, that we have no use for it (yet) and leave this for the future...

Reply:
Also, I dislike `ptype` as name for particles and would try for something better.

Reply:
For consistency with the common lattices, maybe a better name for `L` would be `extent`? (`N` is fine as that's used everywhere consistently.)